### PR TITLE
SDK-332 [C# SDK] Allow oAuth2 token Storage initialization

### PR DESCRIPTION
**Pull request by VladimirLevitskiy**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-332](https://incountry.atlassian.net/browse/SDK-332)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

**Review by robert-par — APPROVED** [review]

File: InCountrySDK/StorageConfig.cs [hunk 1]
```diff
@@ -176,5 +188,19 @@ public StorageConfig UseEndPointFromEnv()
             EndPoint = Environment.GetEnvironmentVariable(ParamEndpoint);
             return this;
         }
+
+        /// <summary>
+        /// Optional. For using of a previously acquired oAuth token for OAuth2 authorization.
```
Comment (RommeroInCountry) [inline InCountrySDK/StorageConfig.cs:193]: For using ... -> Used to set oAuth token manually.

File: InCountrySDK/StorageConfig.cs [hunk 2]
```diff
@@ -95,6 +100,11 @@ public class StorageConfig
         /// </summary>
         public CryptoProvider CryptoProvider { get; set; }
 
+        /// <summary>
+        /// Optional. Function for acquiring of oAuth2 tokens for OAuth2 authorization
```
Comment (RommeroInCountry) [inline InCountrySDK/StorageConfig.cs:104]: Used to acquire oAuth2 token.

File: README.md [hunk 3]
```diff
@@ -167,6 +172,22 @@ var config = new StorageConfig(environmentId, clientId: clientId, clientSecret:
 var storage = Storage.NewStorage(config);
 ```
 
+The SDK also allows using previously acquired oAuth tokens if needed or pass token getting function. In this mode SDK is not responsible for oAuth a token renewal, and it should be done by SDK user himself.
```
Comment (RommeroInCountry) [inline README.md:175]: https://github.com/incountry/sdk-nodejs#oauth-options-configuration please consult this section in Node SDK for proper wording

File: README.md [hunk 4]
```diff
@@ -167,6 +172,22 @@ var config = new StorageConfig(environmentId, clientId: clientId, clientSecret:
...
+
+Below you can find the example of how to specify OAuth token while creating a Storage instance:
+```c#
+//pass a constant token
+var oauthToken = YourGetTokenFunction();
+var config = new StorageConfig(environmentId: "<environment_id>")
+                .SetOauthToken(oauthToken);
+var storage = Storage.NewStorage(config);
+
+//pass an external acquiring of oAuth2 tokens for OAuth2 authorisation
```
Comment (RommeroInCountry) [inline README.md:185]: pass a function returning oAuth2 token

**Review by RommeroInCountry — APPROVED** [review]

File: InCountrySDK/StorageConfig.cs (hunk 1 above)
Comment (VladimirLevitskiy) [reply InCountrySDK/StorageConfig.cs:193]: done at https://github.com/incountry/sdk-csharp/pull/84

File: InCountrySDK/StorageConfig.cs (hunk 2 above)
Comment (VladimirLevitskiy) [reply InCountrySDK/StorageConfig.cs:104]: done at https://github.com/incountry/sdk-csharp/pull/84

File: README.md (hunk 4 above)
Comment (VladimirLevitskiy) [reply README.md:185]: done at https://github.com/incountry/sdk-csharp/pull/84

File: README.md (hunk 3 above)
Comment (VladimirLevitskiy) [reply README.md:175]: The original sentences were modified for C# at https://github.com/incountry/sdk-csharp/pull/84

**Pull request merged**